app/controllers/api/users_controller.rb

```diff
@@ -2,98 +2,51 @@ class Api::UsersController < ApplicationController
   skip_before_action :authenticate_request, only: %i[login register]
 
   def login
-    # TODO: Don't use user params directly, always sanitize user input!
-    # Here: use user_params instead of params
-    authenticate params[:name], params[:password]
+    if authenticate(user_params[:name], user_params[:password])
+      render json: {
+        access_token: @token
+      }, status: :ok
+    end
   end
 
-	def index
-    # TODO: I guess this is a placeholder for now, right?
-		i = {"user" => "Hello world!"}
-		render json: i
-	end
-
   def register
     @user = User.create(user_params)
-    if @user.save
-      response = { message: 'User created successfully' }
+    if authenticate(user_params[:name], user_params[:password])
       render json: {
-        message: response,
-        # TODO: What happens, if the AuthenticateUser call fails?
-        access_token: AuthenticateUser.new(user_params[:name], user_params[:password]).call
-      },
-      status: :created
+        access_token: @token
+      }, status: :ok
     else
       render json: @user.errors, status: :bad
     end
   end
 
-  # TODO: This method should be private
+  def connect
+    @current_user.pending!
+    if User.where(connection_status: :pending).count > 1
+      @other = User.where(connection_status: :pending).where.not(id: @current_user.id).first
+      Connect.new(@other, @current_user).chat
```

Comment:
In both cases, you should return a correct JSON response. See your API documentation, or update it if you do not foresee the return of anything here.